### ISSUE TITLE
Messages are not received if Signal app is not open

### ISSUE THREAD

**Issue by patterchon** [signalapp/Signal-Android#10065]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
I dont get notified about new messages - new messages arrive only when I open the Signal App.
(Some messages to my contacts are not delivered too - perhaps because they also dont get new messages when they dont open the signal app). 

### Steps to reproduce

**Actual result:** I dont get messages / I am not notified about new messages.
**Expected result:** When a new message is sent to me I receive a notification (even if my Signal app is not open).

### Device info
**Device:** Nokia 8.1
**Android version:** 10 (Patches 2020-09-01)
**Signal version:** 4.71.5 (I use Signal app without Google Play services)

### Link to debug log
not possible

**Comment by Nisc3d** [comment]
Have you tried to disable all battery optimizations or data saving settings for Signal already?

**Comment by patterchon** [comment]
Yes - battery optimization and data saving are already disabled for Signal

**Comment by Nisc3d** [comment]
Can you also post a debug log? That would help the developers immensely. You mentioned that you don't have Google play services. If Signal detects it, it will use a websocket connection to get notifications working. Normally it will show a permanent notification because of this. Do you have this notification?

**Comment by patterchon** [comment]
Attached the debug log:
https://debuglogs.org/805d294054c834715bed47d6a18a589cf381a97ef6b0fb5250f7d97690065b97

I used Signal without Google Play Services for a long time now without the described problem. But I didnt ever recognize a notification about web socket connection - how can I check / find this? Is it anywhere in the signal settings?

**Comment by Nisc3d** [comment]
I'm using signal with play services myself, so I only know about this from other people, like issue #6561.

**Comment by bme13** [comment]
Using Signal without FCM myself as well on a test device and that seems to work correct. I have a constant notification visible but as I understood on recent Android versions one can hide that. Not sure about that though. Checked your logfile with one of mine and as far as I can understand things are configured the same. Especially the next line:
```
2020-10-06 22:34:49.640 MESZ D IncomingMessageObserver: Network: true, Foreground: false, FCM: false, Censored: false, Registered: true, Websocket Registered: true
```
Seems to indicate Signal detected that there is no FCM available and Websockets are used.
From the log I also see that Signal is activated every 55s to check if there are new messages so that also seems to be ok. One thing I noticed that after some time of repeating this 55s cycle all of a sudden the network seems to be down. See e.g.
```
2020-10-06 23:01:33.732 MESZ W AlarmSleepTimer: Setting alarm to wake up in 55000ms.
2020-10-06 23:01:33.790 MESZ W IncomingMessageObserver: Application level read timeout...
2020-10-06 23:01:33.793 MESZ D IncomingMessageObserver: Network: true, Foreground: false, FCM: false, Censored: false, Registered: true, Websocket Registered: true
2020-10-06 23:01:33.793 MESZ D IncomingMessageObserver: Reading message...
2020-10-06 23:01:34.380 MESZ D WebSocketConnection: onMessage() -- response received, but no listener
2020-10-06 23:01:34.381 MESZ D WebSocketConnection: onMessage() -- response received, but no listener
2020-10-06 23:02:14.460 MESZ W WebSocketConnection: onFailure()
```
I don't know what caused that.
One other thing I've noticed is that there seem to be 2 alarmtimers but I'm not sure if that's a problem.
```
2020-10-06 23:00:37.744 MESZ W AlarmSleepTimer: Setting alarm to wake up in 55000ms.
2020-10-06 23:00:37.745 MESZ W AlarmSleepTimer: Setting alarm to wake up in 55000ms.
```
I'm not sure if all of this is of any help but hope it points someone in the right direction for a solution.

**Comment by patterchon** [comment]
Since some days and also after the Signal app update to version 4.72.5 messages have been delivered without opening the app, too. I will watch the behaviour for some more days and will give feedback here ...

**Comment by albirs** [comment]
I have similar problems. I'm on website apk 4.72.5 without Google Play Services. I have asking the person after chatting. Seems when the other person is switching to an other app and Signal is not in front any more, messages are not passing through at the moment, even still when i'm writing... Other person has the PlayStore-Version

**Comment by chaoscommander** [comment]
Using 4.73.4 from Play Store I have the same problem. Now updating to 4.74.3, will return in a few days to tell you if it stopped.

**Comment by emmatebibyte** [comment]
Using version 4.75.4 installed from Google Play using Aurora Store (play services installed), I'm having the same issue.

**Comment by chaoscommander** [comment]
The issue persists with version 4.78.5.

**Comment by MorNando** [comment]
I can confirm this is happening on my brothers phone. It is making the app unusable for him

**Comment by gabrivera** [comment]
I have already posted the following text [here](https://github.com/signalapp/Signal-Android/issues/10276), but it may help in this thread as well, given both issues are similar.

> In case y'all have not read this official Signal support article, it may help! I can't test it since I sadly don't have an Android device, but my friends do! And I'll have them try it out, since we have basically gone back to WhatsApp 😕 because some of them weren't getting notifications.
> 
> Strangely, a friend of mine just got a notification from me. It's strange because he hadn't received a single notification/message from me ever since he installed Signal a couple of months ago. He used to have to open the app to see my messages. And now that he finally got a couple of those messages, neither of us know what happened! Anyway, I hope the article helps, and I'm glad I had a reason to post, since this is a massive issue for me. We (my friends and I) basically had to stop using this software because of this bug.

**Comment by greyson-signal** [comment]
Consolidating this into #8692

**Comment by NullDepot** [comment]
Hey everyone. It seems like this issue has been closed and consolidated into #8692 (which is a completely different issue).

Has anyone managed to solve the issue yet? I'm having the same problems on my Android phone running GrapheneOS (no Google Play services).

I'll link my debug file here too since it may help someone else in the future: https://debuglogs.org/7a95ff663fb26e43907a62ea9f69e1d834ec2508793edcd75ca787cea21b0499
